library/java/org/chromium/net/impl/CronetUrlRequest.java

```diff
@@ -574,11 +574,12 @@ private void fireOpenConnection() {
     if (!hasContentType && mUploadDataStream != null) {
       throw new IllegalArgumentException("Requests with upload data must have a Content-Type.");
     }
-    String protocol =
-        isHttp2Enabled && url.getProtocol().equalsIgnoreCase("https") ? "http2" : "http1";
 
-    headers.computeIfAbsent("x-envoy-mobile-upstream-protocol", unused -> new ArrayList<>())
-        .add(protocol);
+    if (isQuicEnabled) {
+      headers.computeIfAbsent("x-envoy-mobile-upstream-protocol", unused -> new ArrayList<>())
+          .add("http3");
```

Comment:
If this header is missing, to which cluster does the request get routed? I guess base_h2 which then has ALPN enabled to it *can* speak HTTP/1, if needed?

Reply:
It will go to _base_ cluster.
With this change, no requests will go to _base_h2_ cluster. In fact, I think for our use case, we could eventually delete the base_h2 cluster since we won't do forced h2 connections.